lib/image-stream/utils.js

```diff
@@ -36,8 +37,22 @@ exports.getArchiveMimeType = (file) => {
   // `archive-type` only needs the first 261 bytes
   // See https://github.com/kevva/archive-type
   const MAGIC_NUMBER_BUFFER_START = 0;
-  const MAGIC_NUMBER_BUFFER_END = 261;
-  const chunk = readChunk.sync(file, MAGIC_NUMBER_BUFFER_START, MAGIC_NUMBER_BUFFER_END);
+  const MAGIC_NUMBER_BUFFER_LENGTH = 261;
 
-  return _.get(archiveType(chunk), [ 'mime' ], 'application/octet-stream');
+  return Bluebird.using(fs.openAsync(file, 'r').disposer((fileDescriptor) => {
+    return fs.closeAsync(fileDescriptor);
+  }), (fileDescriptor) => {
+    const BUFFER_START = 0;
+    const chunk = new Buffer(MAGIC_NUMBER_BUFFER_LENGTH);
+
+    return fs.readAsync(
+      fileDescriptor,
+      chunk,
+      BUFFER_START,
+      MAGIC_NUMBER_BUFFER_LENGTH,
+      MAGIC_NUMBER_BUFFER_START
```

Comment:
Looking at https://nodejs.org/api/fs.html#fs_fs_read_fd_buffer_offset_length_position_callback I guess we could use `null` here, and get rid of `MAGIC_NUMBER_BUFFER_START`.

Also, IMHO `MAGIC_NUMBER` usually indicates a complete number that's being read and matched against exactly. Which isn't actually what we're doing here, it's more of a heuristic-match than an exact-match. So maybe something like `ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH` would be better than `MAGIC_NUMBER_BUFFER_LENGTH` ?